babi/file.py

```diff
@@ -524,20 +524,25 @@ def _indent_selection(self, margin: Margin) -> None:
         assert self.selection.start is not None
         sel_y, sel_x = self.selection.start
         (s_y, _), (e_y, _) = self.selection.get()
+        tab_string = self.buf.tab_string
+        tab_size = len(tab_string)
         for l_y in range(s_y, e_y + 1):
             if self.buf[l_y]:
-                self.buf[l_y] = ' ' * self.buf.tab_size + self.buf[l_y]
+                self.buf[l_y] = tab_string + self.buf[l_y]
                 if l_y == self.buf.y:
-                    self.buf.x += self.buf.tab_size
+                    self.buf.x += tab_size
                 if l_y == sel_y and sel_x != 0:
-                    sel_x += self.buf.tab_size
+                    sel_x += tab_size
         self.selection.set(sel_y, sel_x, self.buf.y, self.buf.x)
 
     @edit_action('insert tab', final=False)
     def _tab(self, margin: Margin) -> None:
-        n = self.buf.tab_size - self.buf.x % self.buf.tab_size
+        tab_string = self.buf.tab_string
         line = self.buf[self.buf.y]
-        self.buf[self.buf.y] = line[:self.buf.x] + n * ' ' + line[self.buf.x:]
+        n = len(tab_string) - self.buf.x % len(tab_string)
+        self.buf[self.buf.y] = (
+            line[:self.buf.x] + tab_string + line[self.buf.x:]
+        )
```

Comment:
maybe I'm reading this wrong but I don't see how this still works correctly.

For example, if we're at position 1 and we press tab it should insert 3 spaces and not 4

Reply:
ah I see now, I'll improve the test: #93

Reply:
if you rebase it should show the test failure